### ISSUE TITLE
no-fallthrough: accept falls-through comments from inside if-else blocks

### ISSUE THREAD

**Issue by mgol** [eslint/eslint#6741]
**The version of ESLint you are using.**
3.1.1

**The problem you want to solve.**
I want to have the `no-fallthrough` rule enabled but sometimes still be able to fall through. This is now handled via the `// falls through` comment but it doesn't work if you use `if-else` and want to fall through only in one of those cases.

This came up while trying to migrate the [AngularJS](https://github.com/angular/angular.js/) repo from JSHint & JSCS to ESLint. This is the relevant code:
https://github.com/angular/angular.js/blob/83bc2479675fd1e0e023c5be1dbcbeb2e2932c6d/src/ngResource/resource.js#L655-L688
A simplified version of this case would be:

``` js
switch (x) {
  case 1:
    if (y) {
      break;
    } else {
      // falls through
    }
  default:
    break;
}
```

**Your take on the correct solution to problem.**
I'd expect the rule to look into `if-else` blocks and accept the code if in each of the branches there is either `break` or the  `// falls through` comment.

**Comment by platinumazure** [comment]
I believe you can avoid this (at least in your simplified case) by removing your else block:

```js
switch (x) {
  case 1:
    if (y) {
      break;
    }
    // falls through
  default:
    break;
}
```

**Comment by mgol** [comment]
The actual code is more complicated and the `else` block is not empty. See the linked Angular example.

**Comment by nzakas** [comment]
@mgol can you include a small repro case that shows the problem you're having now and what ESLint is doing?

**Comment by alberto** [comment]
ping @mgol

**Comment by mgol** [comment]
Sorry for the delay.

A simplified test case would be as follows:
```js
/* eslint no-fallthrough: error */
switch (arguments.length) {
  case 1:
    if (a) {
      if (b) {
        ret = a + b;
        break;
      }

      ret = a;
      // falls through
    } else {
      ret = true;
      break;
    }
  default:
    process(a);
}
```
I have to insert an additional `// falls through` comment just before `default:` but I don't want to do that - I want to explicitly assert that I'm falling through in the first `if` provided that the second condition is not met and in no other case.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
